src/core/mesh/qgsmeshlayer.h

```diff
@@ -182,18 +183,24 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
     const QgsMesh *nativeMesh() const SIP_SKIP;
 
     /**
-     * Returns triangular mesh (NULLPTR before rendering)
+     * Returns triangular mesh (NULLPTR before rendering).
      *
+     * If the parameter triangleSize is provided, among the base triangular mesh
+     * and the simplified triangular meshes, the one returned is which has the average triangle size just greater than triangleSize.
+     * The size of a triangle is the maximum between the height and the width of the triangle bounding box
+     * For default parameter (=0), it returns base triangular mesh.
+     * \param minimumTriangleSize is the average size criteria in canvas map units
+     * \returns triangular mesh
+     * \note triangular size added in QGIS 3.14
      * \note Not available in Python bindings
      */
-    QgsTriangularMesh *triangularMesh() SIP_SKIP;
+    QgsTriangularMesh *triangularMesh( double minimumTriangleSize = 0 ) const SIP_SKIP;
```

Comment:
What happens with owernship here? Is it a factory?

Reply:
keep ownership, added in the doc

Reply:
thanks!